docs/tutorials/clojure-guide.md

```diff
@@ -0,0 +1,225 @@
+is one of the supported servers, you can install it using <kbd>M-x</kbd> `lsp-install-server` <kbd>RET</kbd> `clojure-lsp`. 
+After installed, `lsp-mode` will automatically initialize it when opening Clojure files.
+
+### Manually
+
+Check [here](https://github.com/clojure-lsp/clojure-lsp#installation) for more information on how to build it manually.
+
+For a custom path, you can set the path of the server location with:
+
+```elisp
+(setq lsp-clojure-custom-server-command '("bash" "-c" "/path/to/clojure-lsp"))
+```
+
+## LSP features
+
+`lsp-mode` has a lot of features and integrations, this guide will show the most important ones 
+that I think it's important on Clojure development. You can check all the features on the 
+official [lsp-mode documentation](https://emacs-lsp.github.io/lsp-mode/)
+
+For more info on `lsp-mode` features, check [here](https://emacs-lsp.github.io/lsp-mode/page/main-features/).
+
+### Syntax check
+
+It checks for errors and warnings on your code while you write it, it uses [clj-kondo](https://github.com/clj-kondo/clj-kondo), 
+another amazing library that lints Clojure code, in the back-end so it's not required to install `clj-kondo` since `clojure-lsp` will
+use it automatically.
+
+It will also check for public unused functions/variables which is something that, as far as I know, does not exist in 
+any other Clojure library/package:
+
+![syntax-checker](images/clojure-syntax-checker.png "syntax-checker")
+
+### Find definition/references
+
+You can find a function definition with `lsp-find-definition` or find all the references to that 
+symbol, variable or function with `lsp-find-references` or `lsp-ui-peek-find-references` if 
+[lsp-ui](https://emacs-lsp.github.io/lsp-ui/) is available. 
+
+![find-references](images/clojure-find-references.gif "find-references")
+
+<details>
+<summary>Doom emacs users</summary>
+
+You may need to remove the lookup handlers conflict from `cider` and/or `clj-refactor` if you want to use this LSP feature.
+
+```elisp
+(use-package! cider
+  :after clojure-mode
+  :config
+  (set-lookup-handlers! 'cider-mode nil))
+  
+(use-package! clj-refactor
+  :after clojure-mode
+  :config
+  (set-lookup-handlers! 'clj-refactor-mode nil))
+```
+
+</details>
+
+### Code lens
+
+LSP supports code lens that are texts that server may want to present in client IDE. `clojure-lsp` 
+use it for references count of a function/variable, they are disabled by default so you need to manually enable via:
+
+```elisp
+(setq lsp-lens-enable t)
+```
+
+![reference code-lens](images/clojure-code-lens.png "reference code-lens")
+
+### Call hierarchy
+
+This is a feature recently introduced in LSP 3.16, which servers should return the call hierarchy of a function/method.
+As of now, `clojure-lsp` supports only incoming call hierarchy:
+
+![incoming call-hierarchy](images/clojure-call-hierarchy.png "incoming call-hierarchy")
```

Comment:
In American English, people don't hyphenate compound adjectives as often as they should, so this will probably be written by many people as "incoming call hierarchy". It should be written as "incoming-call hierarchy", since it is a hierarchy of incoming calls.

``` suggestion
![incoming-call hierarchy](images/clojure-call-hierarchy.png "incoming-call hierarchy")
```

Reply:
Why wouldn't it be an incoming hierarchy of calls? `incoming call-hierarchy` sounds better to me, but I am not a native speaker.